src/main/java/com/google/code/or/OpenReplicator.java

```diff
@@ -248,6 +255,21 @@ public void setBinlogEventListener(BinlogEventListener listener) {
 		this.binlogEventListener = listener;
 	}
 
+	protected void setupChecksumState() throws Exception {
+		final Query query = new Query(this.transport);
+
+		try {
+			List<String> cols = query.getFirst("SELECT @@global.binlog_checksum");

```

Comment:
but yes, you're correct, it's the only place in which a replica needs to perform a query on the master.  it's an odd API if you ask me (why not just extend the DUMP_BINLOG command?  Or make a v2 command?), but mysql seems to be leaning towards hacky-but-less-invasive solutions these days.